W-15308521 multiruntime implement duke (#292)
mulesoft/docs-code-builder: modules/ROOT/pages/int-versions.adoc

> = Version Settings for Mule, Java, and Connectors

Author: this is the main file to review

//used in procedures:
:open-project: Open your implementation or integration project. 

Set Mule runtime engine (Mule), Java, and connector versions to use in your projects.

* <<default-versions, Default versions>>: Set default Mule and Java versions to display in version drop-down menus. 

* Project-level versions: Set versions to use in a specific project.
** For Mule and Java versions, see <<project-version-mule-java>>.
** For connector versions, see <<project-versions-connectors>>.  

[[default-versions]]
== Select Default Mule and Java Versions

Select the default Mule and Java versions for new implementation and integration projects. The defaults appear in drop-down menus from which you select a version, for example, when creating a project. 

> Versions that are not present in the IDE append a message similar to *downloads upon project creation*. Versions that do not support a feature typically append a message, such as *AsyncAPI implementation unsupported* for Mule 4.4. 

Reviewer: ```suggestion
Versions not present in the IDE append a message similar to *downloads upon project creation*. Versions that don't support a feature typically append a message, such as *AsyncAPI implementation unsupported* for Mule 4.4. 
```

You can set a different version at the project level (see <<project-version-mule-java>>). 

[[set-defaults-from-palette]]
=== Select Defaults from the Command Palette

To select the default Mule and Java versions from the command palette:

//step: open command palette:
include::partial$acb-reusable-steps.adoc[tags="open-command-palette"]

> . Provide the following command:

Author: ```suggestion
. Run the following command:
```

> . Provide the following command:

Reviewer: 'run' or 'execute' the following command?

include::partial$acb-reusable-steps.adoc[tags="mule-settings-nav-options"]
//TODO: VERIFY NAME CHANGE JDK -> Java Versions
. Find *Mule › Application: Default Mule Runtime And Java Versions* in the open tab.
. Click *Edit defaults through command palette*. 
+
include::partial$acb-reusable-steps.adoc[tags="set-versions-manually"]
//step: select versions from command palette:
include::partial$acb-runtime-java.adoc[tags="runtime-java-version-selection"]

[[project-version-mule-java]]
== Select Mule and Java Versions for a Project

Set Mule or Java versions for a specific implementation or integration project from *Project Properties*. You can change versions at any time, but it is important to test your projects with the versions you intend to deploy. 

Version settings are important in several scenarios:

* Updating a project to use more current versions
* Switching to a supported version, for example, for a project that you import into Anypoint Code Builder that uses an unsupported version
* Adding a version of Mule or Java that is missing from your project
* Selecting versions of Mule or Java that are compatible with each other

> When a Mule or Java version is unset for a project, the IDE's canvas and notifications provide a *Set version* button that opens the *Project Properties* tab. You can also navigate directly to the *Project Properties* tab. 

Reviewer: ```suggestion
When a Mule or Java version is unset for a project, the IDE's canvas and notifications show a *Set version* button that opens the *Project Properties* tab. You can also navigate directly to the *Project Properties* tab. 
```
Optional suggestion.

Author: I'm concerned about using "show," as I'm not sure if it's considered a word affected by accessibility issues.

Author: Same goes in other contexts for words like "see," "looks like," etc. So I'm trying to use alternative language in such cases when I can.

include::partial$acb-runtime-java.adoc[tags="open-project-properties-menu"]
. In *Project Properties*, select the versions to use in this project.
+
image::int-project-properties.png["Setting Mule and Java versions in Project Properties tab"]

[[project-versions-from-palette]]
=== Select Mule and Java Versions from the Command Palette

//open project local attribute:
. {open-project}
//step: open command palette:
include::partial$acb-reusable-steps.adoc[tags="open-command-palette"]
//step: open project properties tab from command palette
include::partial$acb-runtime-java.adoc[tags="open-project-properties-command"]
//TODO: VERIFY Get a Runtime EVERYWHERE
//TODO: PROVIDE SCREENSHOTS
//step: select versions from command palette:
include::partial$acb-runtime-java.adoc[tags="runtime-java-version-selection"]

[[project-versions-connectors]]
== Select Connector Versions for a Project

Set connector versions for a specific implementation or integration project from *Project Properties*. You can change versions at any time, but it is important to test your projects with the versions you intend to deploy. 

* <<connector-version-from-properties>>
* <<connector-version-from-palette>>
 

> //TODO: WILL THIS GET IMPLEMENTED?
> //TODO: RECOMMENDATIONS?
> //TODO: WHAT IF EMPTY?

Author: ```suggestion
//TODO: RECOMMENDATIONS? 
//TODO: WHAT IF EMPTY? an empty notification will appear in the UI -- no need to add image or details in doc
```

[[connector-version-from-properties]]
=== Select Connector Versions from the Project Properties Tab

//open project local attribute:
. {open-project}
//step: open project properties tab from Project Properties menu item
include::partial$acb-runtime-java.adoc[tags="open-project-properties-properties"]
//step: open project properties tab
include::partial$acb-runtime-java.adoc[tags="open-project-properties-menu"]
. Click *Connectors* from the *Project Properties* side bar. 

> //TODO: P3: ADD SCREENSHOT AND EXAMPLES

Author: TODO: SCREENSHOT WHEN AVAILABLE W/O having to see "JDK" in the side bar

. Select connector versions. 

[[connector-version-from-palette]]
=== Select Connector Versions from the Command Palette

. {open-project}
//step: open command palette:
include::partial$acb-reusable-steps.adoc[tags="open-command-palette"]
//step: open project properties tab from command palette
include::partial$acb-runtime-java.adoc[tags="open-project-properties-command"]
. Click *Connectors* from the *Project Properties* side bar. 

> //TODO: P3: ADD SCREENSHOT AND EXAMPLES

Author: TODO: same SCREENSHOT as above WHEN AVAILABLE W/O having to see "JDK" in the side bar
